src/docs/upgrading.dox

```diff
@@ -1,5 +1,67 @@
 /*! @page upgrading Upgrading WiredTiger applications
+@section version_310 Upgrading to Version 3.1.0
+<dl>
+
+<dt>WiredTiger on-disk log file format change</dt>
+<dd>
+The WiredTiger on-disk file format for write-ahead log files has changed
+as the log file version number was incremented. See
+<a href=https://jira.mongodb.org/browse/WT-4029>WT-4029</a> for details.
+</dd>
+
+<dt>::wiredtiger_open compatibility configuration changes</dt>
+<dd>
+The compatibility setting now takes additional options that can define
+the minimum or maximum required version of existing data files. See
+<a href=https://jira.mongodb.org/browse/WT-4098>WT-4098</a> for details.
+</dd>
+
+<dt>Changed transaction semantics around schema operations</dt>
+<dd>
+WiredTiger does not offer fully transactional create and drop operations.
+We have made some changes to how create and drop are implemented
+if done within the scope of an explicit transaction. If an application
+is relying on particular visibility/atomicity guarantees around table
+create or drop, care should be taken when upgrading. See
+<a href=https://jira.mongodb.org/browse/WT-3964>WT-3964</a> for details.
+</dd>
+
+<dt>On-disk format change for metadata</dt>
+<dd>
+There was a change to the content stored in the WiredTiger owned metadata
+files, which means metadata created or updated by this version of WiredTiger
+is not compatible with earlier versions. See
+<a href=https://jira.mongodb.org/browse/WT-3905>WT-3905</a> for details.
+</dd>
+
+<dt>::wiredtiger_open eviction_checkpoint_target configuration change</dt>
+<dd>
+The default value for the \c eviction_checkpoint_target configuration
+setting was changed from 5 to 1. This was due to a change in the algorithm
+to focus more on reducing dirty content in low throughput workloads. See
+<a href=https://jira.mongodb.org/browse/WT-4111>WT-4111</a> for details.
+</dd>
+
+<dt>Implement a per-session cursor cache</dt>
+<dd>
+WiredTiger now holds a cache of recently closed cursors in each
+session. This improves performance for applications that open and
+close cursors frequently, but increases memory overhead. The cache
+is enabled by default, but can be disabled. See
+<a href=https://jira.mongodb.org/browse/WT-1228>WT-1228</a> for details.
+</dd>
+
+<dt>::wiredtiger_open cache configuration changes</dt>
+<dd>
+The cache configuration options \c eviction_checkpoint_target, \c
+eviction_dirty_target, \c eviction_dirty_trigger, \c eviction_target and \c
+eviction_trigger have changed. The options can now take an absolute size. It would
+be a percentage of the cache size if the value is within the range of 0 to 100
+or an absolute size when greater than 100. This API change is compatible with
+existing usage.
```

Comment:
With each of above sections referring a WT ticket for details, should we add the link of WT-3632 here for consistency?